### PR TITLE
Fixes: #14567 - Export current view of IP Addresses

### LINKED ISSUE #14567: Export current view of IP Addresses
### Deployment Type

Self-hosted

### NetBox Version

v3.6.7

### Python Version

3.8

### Steps to Reproduce

Tested with demo.netbox.dev
1. Select IPAM -> IP Addresses (on the top Results shows 183)
image
2.  Quick search 172.16  (Results stays on 183)
image
4. Export Current view 
image

### Expected Behavior

Shouldn't it be the case that when filtering by addresses, the number of results is reduced and accordingly the export of the current view only contains the filtered addresses?

### Observed Behavior

When exporting, all addresses are always exported.

### PR DESCRIPTION
**Pull request by Julio-Oliveira-Encora**
### Fixes: #14567 

It added a script in "table_controls_htmx.html" to add the parameters to the URL (changing the htmx behavior).

Another script was created in "export.html" to append the parameters to the HREF to export the current view. 

It was tested exporting devices, IP Address, and IP Range, with and without a quick search.

### CODE REVIEW

File: netbox/templates/inc/table_controls_htmx.html [hunk 1]
```diff
@@ -2,22 +2,49 @@
...
+    function getParams() {
+        const quickSearchParameters = htmx.values(htmx.find("#quick_search"));
+        let link = document.getElementById("current_view")
+        if (quickSearchParameters != null) {
+            const search_parameter = `q=${quickSearchParameters.q}`
+            link = link.href + "&" + search_parameter
+        }
+        document.getElementById("current_view").setAttribute("href", link)
+    }
+</script>
```
Comment (jeremystretch) [inline netbox/templates/inc/table_controls_htmx.html:50]: All scripting should be written in TypeScript and should live in a module under `project-static/src/` with the rest of the frontend scripting.

**Review by jeremystretch — CHANGES_REQUESTED** [review]

File: netbox/project-static/src/search.ts [hunk 2]
```diff
@@ -9,36 +10,59 @@ function quickSearchEventHandler(event: Event): void {
...
+ * @param event
+ */
+
+function handleQuickSearchParams(event: Event): void {
+  console.log('getParams');
+  const quickSearchParameters = event.currentTarget as HTMLInputElement;
+  const link = document.getElementById('current_view') as HTMLLinkElement;
+  if (quickSearchParameters != null) {
+    const search_parameter = `q=${quickSearchParameters.value}`;
+    const linkUpdated = link?.href + '&' + search_parameter;
```
Comment (jeremystretch) [inline netbox/project-static/src/search.ts:32]: This appears to append successive queries, e.g. `?q=foo&q=bar`. Any previous query should be replaced by the new one.

**Review by jeremystretch — CHANGES_REQUESTED** [review]
Getting closer!

File: netbox/project-static/src/search.ts [hunk 3]
```diff
@@ -9,36 +10,59 @@ function quickSearchEventHandler(event: Event): void {
...
-    quicksearch.addEventListener("search", quickSearchEventHandler, {
-      passive: true
-    })
+    quicksearch.addEventListener('keyup', quickSearchEventHandler, {
+      passive: true,
+    });
+    quicksearch.addEventListener('search', quickSearchEventHandler, {
+      passive: true,
+    });
+    quicksearch.addEventListener('change', handleQuickSearchParams);
```
[the remark this answers is not in the document]
Comment (Julio-Oliveira-Encora) [reply netbox/project-static/src/search.ts:51]: I created a function to clear the query parameters and used it in the "quickSearchEventHandler"; this way, when the Quick Search field is cleaned up, the parameters are also removed.

The same function is used when the search is changed, cleaning the current parameters before adding the new ones.

File: netbox/project-static/src/search.ts [hunk 4]
```diff
@@ -25,20 +55,26 @@ export function initQuickSearch(): void {
...
+        async () => {
+          const search = new Event("search");
+          quicksearch.value = "";
+          await new Promise(f => setTimeout(f, 100));
+          quicksearch.dispatchEvent(search);
+        },
+        {
+          passive: true,
+        },
+      );
```
Comment (jeremystretch) [inline netbox/project-static/src/search.ts:77]: Please revert any changes that aren't pertinent to the new functionality to limit the scope of PR reviews. (If cleanup is needed, we can tackle those items in a separate housekeeping issue.)

File: netbox/project-static/src/search.ts [hunk 5]
```diff
@@ -11,12 +11,42 @@ function quickSearchEventHandler(event: Event): void {
...
+  const link = document.getElementById("current_view") as HTMLLinkElement;
+  let linkUpdated = link?.href.split("&")[0];
+  link.setAttribute("href", linkUpdated);
+}
+
+/**
+ * Update the Export View link to add the Quick Search parameters.
+ * @param event
+ */
+
```
Comment (jeremystretch) [inline netbox/project-static/src/search.ts:34]:
```suggestion
```

File: netbox/project-static/src/search.ts [hunk 6]
```diff
@@ -11,12 +11,42 @@ function quickSearchEventHandler(event: Event): void {
...
       clearbtn.classList.remove("invisible");
     }
   }
 }
 
+/**
+ * Clear the existing search parameters in the link to export Current View.
+ */
+function clearLinkParams(): void {
+  const link = document.getElementById("current_view") as HTMLLinkElement;
```
Comment (jeremystretch) [inline netbox/project-static/src/search.ts:25]: Could we give this element a slightly more descriptive name, to help prevent confusion?
```suggestion
  const link = document.getElementById("export_current_view") as HTMLLinkElement;
```

**Review by jeremystretch — CHANGES_REQUESTED** [review]

File: netbox/project-static/src/search.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jeremystretch) [reply netbox/project-static/src/search.ts:60]: This does not appear to be working. After entering a search term in the quicksearch field and clicking the clear button, the term is still attached to the export link.

**Review by jeremystretch — CHANGES_REQUESTED** [review]

**Review by arthanson — CHANGES_REQUESTED** [review]
I'm still getting the whole table when I'm exporting.  See screenshot, I only have one Address that is in the table, but export I get all.
Monosnap IP Addresses | NetBox 2024-05-30 12-42-52

**Comment by Julio-Oliveira-Encora** [comment]
I tested it. When I exported, I got only one IP address.

ip_addr
export_IP_addr

**Comment by Julio-Oliveira-Encora** [comment]
I checked out to another branch and exported the current view. Then, I went back to the branch that fixed the export current view. Refreshing the page was necessary for it to work correctly.

**Review by jeremystretch — APPROVED** [review]
Thanks @Julio-Oliveira-Encora!

**Pull request merged**